### DISCUSSION TITLE
Demo：NebulaGraph 的知识图谱构建与 Text2Cypher - siwei.io

### DISCUSSION THREAD

**Discussion by giscus** [siwei-io/siwei-io.github.io#27] [Comments]
# Demo：NebulaGraph 的知识图谱构建与 Text2Cypher - siwei.io

https://www.siwei.io/demos/text2cypher/

**Comment by bg-vc** [comment]
请问后续有没有打算详细写一下“利用 LLM 从文档中抽取知识三元组，并存储到 GraphStore”的具体实现？

**Reply by wey-gu** [reply]
会写的哈！

**Comment by lazyloafer** [comment]
请问GraphStore有neo4j的接口吗？

**Reply by wey-gu** [reply]
我引入这个抽象就是允许多个后端的，后边 neo4j 也引入了。

不过现在只有 NebulaGraphStore 支持带有方向的 GraphRAG 和自定义 tag 返回属性集合。

**Reply by lazyloafer** [reply]
我这边有一个neo4j数据库，希望大佬能出一个neo4j导入NebulaGraphStore的教程

**Reply by wey-gu** [reply]
赞，之后我可以搞一个详细教程哈！不过基本思路是

1. 要看一下 neo4j 中图的结构，因为 NebulaGraph 是强 schema 的设计，我们要准备好如何把不同 label node 的属性对应到 NebulaGraph 这边的 label（这边叫作 tags），并且创建好 NebulaGraph 中的 tags。[参考这个文档哈](https://docs.nebula-graph.com.cn/3.6.0/1.introduction/2.data-model/)！
2. 利用 NebulaGraph Exchange 进行数据导入，参考[文档](https://docs.nebula-graph.com.cn/3.6.0/nebula-exchange/use-exchange/ex-ug-import-from-neo4j/)，这里写的配置就是两边点、边类型和属性的对应关系。

注意，因为是利用 exchange 这个工具，不推荐用 docker 部署 NebulaGraph，可以考虑弄一台机器，用 .rpm/.deb的包安装哈！

**Reply by wey-gu** [reply]
可以访问 https://discuss.nebula-graph.com.cn/ 中文论坛，里便也有一些知识，还有微信群可以加哈（好像是从 NebulaGraph 这个公众号里有入口）

**Reply by lazyloafer** [reply]
感谢大佬投喂，我先调研一下哈，期待您的教程！

**Reply by wey-gu** [reply]
> 感谢大佬投喂，我先调研一下哈，期待您的教程！

期待您的探索！

**Comment by lylyone** [comment]
请问大佬，我看自然语言实现写sql,这个大模型是使用的是openai还是离线训练过的大模型呢

**Comment by aqzwy** [comment]
大佬，我想知道langchain的使用版本是多少，我经过llm抽取三元组发现和您的结果并不是一致的，而且在用langchain的是后续使用rag_query 是需要安装nltk，query的结果也是发生了变化
```
Peter Quill is a character in the Marvel Cinematic Universe. He is the leader of the Guardians of the Galaxy, and was fired from the Guardians of the Galaxy 3 project in 2018. He is set to return to the MCU in May 2021.
```

**Comment by xiongsircool** [comment]
大佬你好有一个问题关于这个给图可视化的如果是在前端的话常用的可视化应该用插件或者包吗。我就用过vis.js 但是感觉显示多节点时候会出现一团的情况

**Comment by DreamingJ** [comment]
【问题反馈】这个Demo由于Openai的接口版本更新，现在已经无法正常显示了。
访问时提示信息如下：
APIRemovedInV1: You tried to access openai.Completion, but this is no longer supported in openai>=1.0.0 - see the README at https://github.com/openai/openai-python for the API. You can run `openai migrate` to automatically upgrade your codebase to use the 1.0.0 interface. Alternatively, you can pin your installation to the old version, e.g. `pip install openai==0.28` A detailed migration guide is available here: https://github.com/openai/openai-python/discussions/742

**Reply by DreamingJ** [reply]
希望大佬有时间可以更新一下，感谢！

**Reply by wey-gu** [reply]
感谢提醒，恢复啦

**Comment by lemon-little** [comment]
为什么刷新不出这篇文章的内容？

**Reply by wey-gu** [reply]
可能需要翻墙，背后是 streamlit 哈，似乎会被干扰

**Comment by cementwise** [comment]
请问后续有没有不调openai接口，而是使用离线rag结合graph的内容呢？
